models/web/snowplow_media_player_media_stats.sql

```diff
@@ -43,7 +43,8 @@ with new_data as (
 from {{ ref("snowplow_media_player_base") }} p
 
 where -- enough time has passed since the page_view's start_tstamp to be able to process it as a whole (please bear in mind the late arriving data)
-p.start_tstamp < cast({{ dbt_utils.dateadd('hour', var("snowplow__max_media_pv_window", 10), dbt_utils.current_timestamp_in_utc() ) }} as {{ dbt_utils.type_timestamp() }})
+{{ dbt_utils.dateadd('hour', var("snowplow__max_media_pv_window", 10), 'p.start_tstamp ') }} < cast({{ dbt_utils.current_timestamp_in_utc() }} as {{ dbt_utils.type_timestamp() }})
```

Comment:
Would it make sense to change this to using `p.end_tstamp`, since then we're waiting until enough time has passed since the last event rather than the first event, or does that not make sense?

Reply:
Also you no longer need to cast `{{ dbt_utils.current_timestamp_in_utc() }}` as `{{ dbt_utils.type_timestamp() }}` since it'll already be the `TIMESTAMP` type, but we do need to cast the first part to a timestamp because right now it's a DATETIME and BQ doesn't know how to handle that for some reason

Reply:
Was thinking about it too, but if we calculated it with the p.end_stamp then anything that bots keep alive for long would never make into the stats. However, I would not think bot activity would interfere here or if yes, perhaps it is better to keep those out (?).

Reply:
Yeah but we do quarantine sessions that go on for longer than 3 days in the web package already, so I guess that should be handled by that process. I'm not heavily opinionated on it either way, but I'm just thinking if we limit the sesison duration to 3 days with `snowplow__max_session_days` it feels weird to then give the user another variable to start processing data in the form of `snowplow__max_media_pv_window`, but I don't have super strong opinions about it either way